src/java6/java/org/spongepowered/launch/JavaVersionCheckUtils.java

```diff
@@ -52,4 +52,32 @@ public static void ensureJava8() {
     private static String getCurrentVersion() {
         return System.getProperty("java.version");
     }
+
+    /**
+     * Calculates a double value based on a Java version string such that a
+     * higher version will produce a higher value
+     *
+     * @param version The Java version
+     * @return The double value of the Java version
+     */
+    private static double getVersionValue(String version) {
+        // Get rid of any dashes, such as those in early access versions which have "-ea" on the end of the version
+        if(version.contains("-")) {
+            version = version.substring(0, version.indexOf('-'));
+        }
+        // Replace underscores with periods for easier String splitting
+        version = version.replace('_', '.');
+        // Split the version up into parts
+        String[] versionParts = version.split("\\.");
+        double versionValue = 0;
+        for(int i = 0; i < versionParts.length; i++) {
+            int part = Integer.valueOf(versionParts[i]);
```

Comment:
Should do some more strict checking here, will throw an exception if it's null or not a number. It's not a good idea to crash because a bad `java.version`